query-engine/js-connectors/js/planetscale-js-connector/src/planetscale.ts

```diff
@@ -181,21 +67,67 @@ class PrismaPlanetScale implements Connector {
   private async performIO(query: Query) {
     const { sql, args: values } = query
 
-    try {
-      return await this.driver.client.execute(sql, values)
-    } catch (e) {
-      const error = e as Error & { code: string }
-      
-      if (isConnectionUnhealthy(error.code)) {
-        this._isHealthy = false
-      }
+    return await this.client.execute(sql, values)
+  }
+}
 
-      throw e
-    }
+class PlanetScaleTransaction extends PlanetScaleQueryable<planetScale.Transaction> implements Transaction {
+  constructor(tx: planetScale.Transaction, private txDeferred: Deferred<void>, private txResultPromise: Promise<void>) {
+    super(tx)
+  }
+
+  commit(): Promise<void> {
+    const tag = '[js::commit]'
+    debug(`${tag} committing transaction`)
+    this.txDeferred.resolve()
+    return this.txResultPromise;
+  }
+
+  rollback(): Promise<void> {
+    const tag = '[js::rollback]'
+    debug(`${tag} rolling back the transaction`)
+    this.txDeferred.reject(new RollbackError())
+    return this.txResultPromise;
+  }
+
+}
+
+class PrismaPlanetScale extends PlanetScaleQueryable<planetScale.Connection> implements Connector {
+  constructor(config: PrismaPlanetScaleConfig) {
+    const client = planetScale.connect(config)
+
+    super(client)
+    
   }
+
+  async startTransaction(isolationLevel?: string) {
+    return new Promise<Transaction>((resolve) => {
+      const txResultPromise = this.client.transaction(async tx => {
+        if (isolationLevel) {
+          await tx.execute(`SET TRANSACTION ISOLATION LEVEL ${isolationLevel}`)
+        }
```

Comment:
Here, we don't need to manually submit `BEGIN` due to PlanetScale's `this.client.transaction()` doing it for us?

Reply:
Yes, PlanetScale's `transaction` does `BEGIN`, `COMMIT` and `ROLLBACK` automatically.